Initial suite of samples

zachernuk opened the pull request:
Using the teapot character, demonstrating the API as discussed so far!

zachernuk committed:
Initial suite of samples

Using the teapot character, demonstrating the API as discussed so far.

zachernuk commented:
@mkeblx These are the samples I mentioned!

mkeblx commented on autoplay-loop.html:
```diff
@@ -0,0 +1,36 @@
+    </p>
+
+    <pre>
+&lt;model autoplay alt="A model of a teapot with a hidden creature">
```
No `loop` attribute present here in code sample

mkeblx commented on autorotate.html:
```diff
@@ -0,0 +1,70 @@
+
+    <h2>HTML</h2>
+    <pre>
+&lt;model id="model" alt="A model of a teapot">
```
For these code sections I believe want to also excape the > chars.

mkeblx commented on autorotate.html:
```diff
@@ -0,0 +1,70 @@
+  </body>
+  <script type="module">
+    let initialTransform;
+    await model.ready;
```
Some samples like here use the more ~legacy fact that HTML element with ID `x` shows up `window.x`, while others do not. Can we always explicitly reference the element via a selector?

mkeblx commented on basic-data-url.html:
```diff
@@ -0,0 +1,41 @@
+const model = document.querySelector('model');
+const {src, type} = model.dataset;
+const bytes = await (await fetch(src)).bytes();
+const srcString = URL.createObjectURL(new Blob([bytes], {type}));
```
This is actually a blob URL test case, as using createObjectURL => blob.

Blob URL:
 ```js
const response = await fetch(url);
const blob = await response.blob(); // this should maintain type too
const objectURL = URL.createObjectURL(blob);
```

We should probably have two demo pages, one for data URL, and blob URL.

~data URL:
```js
const response = await fetch(url);
const blob = await response.blob();
const reader = new FileReader();
reader.readAsDataURL(blob);
```

mkeblx commented on currenttime.html:
```diff
@@ -0,0 +1,46 @@
+  </body>
+
+  <script>
+    (async () => {
```
Like other samples, can we use `script type="module"` which has top level await and remove this async wrapper?

mkeblx commented on ibl.html:
```diff
@@ -0,0 +1,56 @@
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Model: Basic test</title>
```
Include IBL in title

mkeblx commented on jsorbit.html:
```diff
@@ -0,0 +1,109 @@
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Model: Orbit test</title>
```
JS-Orbit in title, like H1

mkeblx commented on playbackrate.html:
```diff
@@ -0,0 +1,45 @@
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Model: basic test</title>
```
Playbackrate in title

mkeblx commented on currenttime.html:
```diff
@@ -0,0 +1,46 @@
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Model: basic test</title>
```
currentTime in title like H1

mkeblx commented on boundingbox.html:
```diff
@@ -0,0 +1,43 @@
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Model: basic test</title>
```
boundingBox in title

mkeblx commented on background.html:
No title tag in head

mkeblx reviewed (changes requested):
Great start on some samples 🫖 👍 

A few minor things

zachernuk committed:
Addressing @mkeblx's feedback!

Replaced the `>` characters in the `pre` sections with the escaped character,
Made use of type="module" script tags and omitted the script tags in their respective `pre` sections,
Added and updated the titles,
Renamed the DataURL test to accurately reflect that it's a Blob URL test,
Made sure there is a valid query selector rather than relying on the global-namespaced references by ID.

zachernuk commented:
@mkeblx Thank you so much for catching all that! Take a look again when you can.

mkeblx reviewed (approved):
Great, LGTM!
